### ISSUE TITLE
add-to-cart-variation.js broken when using Content Security Policy w/o unsafe-eval

### ISSUE THREAD

**Issue by valdrinkoshi** [woocommerce/woocommerce#19129]
I'm setting a restrictive Content Security Policy directive for scripts: `"script-src: 'self' 'unsafe-inline'` (note: purposefully omitting `unsafe-eval`).
When the consumer tries to select a variation, the Add to Cart button won't be enabled (because of the javascript error in console):
```
Uncaught EvalError: Refused to evaluate a string as JavaScript because
'unsafe-eval' is not an allowed source of script in the following
Content Security Policy directive: "script-src 'self' 'unsafe-inline'"
```

The WooCommerce frontend uses `wp.template` which seem to require `unsafe-eval` https://core.trac.wordpress.org/ticket/38695#comment:1
This is used in `add-to-cart-variation` to generate new DOM and populate it with variation data:
https://github.com/woocommerce/woocommerce/blob/55b3f339d3120571773522e2adceb92a49af2e11/assets/js/frontend/add-to-cart-variation.js#L213-L218

This is the only frontend spot relying on `wp.template`, would you consider moving away from using it? If not, do you have any suggestions on how to work around this without requiring the user to allow `unsafe-eval`?

**Comment by mikejolley** [comment]
Using wp.template allows custom placeholders to be used. Your PR does not unfortunately.  e.g. `data.variation.customvalue`

https://core.trac.wordpress.org/ticket/38695#comment:1

It seems that ticket is wontfix. I don't think we can move away from `wp.template` without it being breaking at this point. We don't know if extensions are using the custom data tags for other things. 

With that in mind, I don't think we can support this configuration and like core, this should be marked wontfix. But if we ever do refactor this and templates, dropping all libraries (e.g. this, jquery) should happen also.

cc @claudiulodro @kloon @claudiosanches so this has some extra visibility for future reference.

**Comment by valdrinkoshi** [comment]
@mikejolley I've addressed the custom value use case in https://github.com/woocommerce/woocommerce/pull/19496. I did open the PR just as an FYI to people facing the same issue. I couldn't find any other better way to circumvent the problem other than patch it manually and always remember to upload these 2 files when upgrading to a new version of woocommerce :/ 
When in production, the code is minified, so there is no easy way to patch that function. Patching `wp.template` would be another option, but I'm afraid to fall in a rabbit hole if i do that :p
